### ISSUE TITLE
add regex as words in dictionnary

### ISSUE THREAD

**Issue by PaulEcoffet** [yitzchak/linter-spell#41]
Good evening,

I write report in markdown using pandoc references system which are in the format `@article_id`. As the article id is usually a surname followed by a year, it is always taken as misspelled word. I tried to add `@\w*` in the github dictionnary but it did not work. Is it possible to use regex as word? Or to make `@bla` not checked by linter-spell?

The same thing happens for '---------------------------' used as title. for me
```
Introduction
----------------
```
the '---' are considered as a misspelled word. adding '-*' in the dictionnary did not work.

Thank you for your help!

**Comment by dhabyx** [comment]
I agree with Paul, in many texts I use md5 or sha sums, and sometimes is tedious mark all this with ignore option.

**Comment by yitzchak** [comment]
Thanks for the feedback. I'll look into what it would take to accomplish this.

**Comment by Aerijo** [comment]
This is best covered by the grammar package, which should label such constructions as a specific scope. The language specific linter-spell package can then be told to blacklist these scopes from spell checking. I am not sure why this is not already the case, unless this is an odd usecase.

Also, @PaulEcoffet, dictionary entries are covered by hunspell or aspell (whichever you use), so here is not really the place to ask. From my (brief) look at these, it appears they don't support regex. Regex can be used in the grammar files, to scope these constructions.
